regions/shapes/moc/core.py

```diff
@@ -0,0 +1,941 @@
+
+        # Configure a WCS
+        from regions._utils.examples import make_example_dataset
+        config = dict(crpix=(0, 0), crval=(0, 0), cdelt=(-5, 5), shape=(18, 36))
+        dataset = make_example_dataset(config=config)
+
+        # Convert the MOCSkyRegion to a MOCPixelRegion for plotting it
+        moc_px_reg = moc_sky_reg.to_pixel(dataset.wcs)
+
+        # Define matplotlib to plot the MOCPixelRegion
+        import matplotlib.pyplot as plt
+
+        fig, ax = plt.subplots(1, 1, figsize=(10, 10))
+        moc_px_reg.plot(ax, edgecolor='red', lw=1)
+
+        plt.xlim(0, 4)
+        plt.ylim(4, 8)
+        plt.show()
+    """
+    HPY_MAX_NORDER = 29
+
+    def __init__(self, interval_set=None, meta=None, visual=None):
+        interval = IntervalSet() if interval_set is None else interval_set
+        self._interval_set = interval
+        self.meta = meta or RegionMeta()
+        self.visual = visual or RegionVisual()
+        self._repr_params = ('_interval_set',)
+
+    def to_pixel(self, wcs):
+        healpix_cells_d = self.write(format='json')
+
+        X = np.array([])
+        Y = np.array([])
+        for order_str, ipix_l in healpix_cells_d.items():
+            order = int(order_str)
+            # Compute the skycoord boundaries of the HEALPix cells from one order at a time.
+            hp = HEALPix(nside=(1 << order), order='nested', frame=ICRS())
+            vertices_boundaries = hp.boundaries_skycoord(ipix_l, step=1)
+
+            x, y = skycoord_to_pixel(vertices_boundaries, wcs=wcs)
+            if X.size >= 1:
+                X = np.vstack((X, x))
+                Y = np.vstack((Y, y))
+            else:
+                X = x
+                Y = y
+
+        # Convert them to pixel coordinates
+        vertices = PixCoord(X, Y)
+        return MOCPixelRegion(vertices=vertices,
+                              sky_region=self,
+                              meta=self.meta,
+                              visual=self.visual)
+
+    def plot(self):
+        raise NotImplementedError
+
+    def __eq__(self, another_moc):
+        """
+        Test equality between self and ``another_moc``
+
+        Parameters
+        ----------
+        another_moc : `~regions.MOCSkyRegion`
+            The moc object to test the equality with.
+
+        Returns
+        -------
+        result : bool
+            True if the interval sets of self and ``another_moc`` are equal.
+        """
+        if not isinstance(another_moc, MOCSkyRegion):
+            raise TypeError('The object you want to test the equality with is not a MOC but a {0}'
+                            .format(type(another_moc)))
+
+        return self._interval_set == another_moc._interval_set
+
+    @classmethod
+    def from_json(cls, json_moc, meta=None, visual=None):
+        """
+        Create a `~regions.MOCSkyRegion` from a dictionary of HEALPix cell arrays indexed by their order (i.e. in json format).
```

Comment:
The first summary line of a docstring should always be short. In this case e.g. "Create from a JSON representation."

This line is super long and will not appear nicely in the method summary in the API docs.

Please build the HTML docs locally with `python setup.py build_docs` and browse through the API summary list and docstrings to make sure they look OK.